carbonmark-api/src/routes/users/[walletOrHandle]/get.utils.ts

```diff
@@ -43,23 +53,58 @@ const fetchTestnetHoldings = async (params: {
       decimals: 18,
     },
   ];
+
   const balancePromises = TOKEN_INFO.map((tco2) =>
     new Contract(tco2.address, ERC20, provider).balanceOf(params.address)
   );
-  const tco2Balances: bigint[] = await Promise.all(balancePromises);
-  const holdings: Holding[] = tco2Balances.map((balance, i) => {
-    return {
-      amount: balance.toString(),
-      id: `0x_mock_holding_id_${i}`,
+
+  const fetchTco2Balances = Promise.all(balancePromises);
+  const fetchIcrHoldings = sdk.icr.getHoldingsByAddress({
+    id: params.address,
+  });
+
+  const [tco2Balances, IcrBalances] = await Promise.all([
+    fetchTco2Balances,
+    fetchIcrHoldings,
+  ]);
+
+  const tco2Holdings: MultipleTokenStandardType[] = tco2Balances.map(
+    (balance, i) => {
+      return {
+        amount: balance.toString(),
+        id: `0x_mock_holding_id_${i}`,
+        token: {
+          decimals: 18,
+          id: TOKEN_INFO.at(i)?.address ?? "",
+          name: TOKEN_INFO.at(i)?.symbol ?? "",
+          symbol: TOKEN_INFO.at(i)?.symbol ?? "",
+        },
+      };
+    }
+  );
+
+  let icrHoldings: MultipleTokenStandardType[] = [];
+  if (IcrBalances.holder?.exPostAmounts) {
+    icrHoldings = IcrBalances.holder.exPostAmounts.map((item) => ({
+      amount: utils.parseUnits(item.amount, 18).toString(),
+      id: item.exPost.serialization,
       token: {
         decimals: 18,
-        id: TOKEN_INFO.at(i)?.address ?? "",
-        name: TOKEN_INFO.at(i)?.symbol ?? "",
-        symbol: TOKEN_INFO.at(i)?.symbol ?? "",
+        id: item.exPost.project.id,
+        name: item.exPost.project.projectName,
+        symbol:
+          createICRProjectID(item.exPost.serialization) +
+          "-" +
+          item.exPost.vintage,
```

Comment:
You can use javascript templating here:
`'${createICRProjectID(item.exPost.serialization)}-${item.exPost.vintage}'`